src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -68,6 +68,61 @@ def retrieve_electrical_series(nwbfile: NWBFile, electrical_series_name: Optiona
     return electrical_series
 
 
+def _read_hdf5_file(
+    *,
+    file_path: str | Path | None,
+    file: BinaryIO | None = None,
+    stream_mode: Literal["ffspec", "ros3", "remfile"] | None = None,
+    cache: bool = False,
+    stream_cache_path: str | Path | None = None,
+):
+    import h5py
+
+    if stream_mode == "fsspec":
+        import fsspec
+        from fsspec.implementations.cached import CachingFileSystem
+
+        assert file_path is not None, "file_path must be specified when using stream_mode='fsspec'"
+
+        fsspec_file_system = fsspec.filesystem("http")
+
+        if cache:
+            stream_cache_path = stream_cache_path if stream_cache_path is not None else str(get_global_tmp_folder())
+            caching_file_system = CachingFileSystem(
+                fs=fsspec_file_system,
+                cache_storage=str(stream_cache_path),
+            )
+            ffspec_file = caching_file_system.open(path=file_path, mode="rb")
+        else:
+            ffspec_file = fsspec_file_system.open(file_path, "rb")
+
+        hdf5_file = h5py.File(name=ffspec_file, mode="r")
+
+    elif stream_mode == "ros3":
+        assert file_path is not None, "file_path must be specified when using stream_mode='ros3'"
+
+        drivers = h5py.registered_drivers()
+        assertion_msg = "ROS3 support not enbabled, use: install -c conda-forge h5py>=3.2 to enable streaming"
```

Comment:
Is ros3 only through conda-forge. I know most people use anaconda, but most of the other assertion messages say pip install instead. I know you just copied this over from the old assert, but was just curious why we had this there originally?

Reply:
Yep, it requires h5py with that driver which I don't know how else to install. We are kind of beating around the bush about deprecating ros3 as we have not found any advantages over fsspec so far and is hard to test and hard to install.

Reply:
Makes sense. You seem to be in the camp to just remove. Is there anything in favor of keeping it?